api/v1alpha08/sonataflow_persistence_types.go

```diff
@@ -51,10 +51,13 @@ type PersistenceOptionsSpec struct {
 	// +optional
 	PostgreSQL *PersistencePostgreSQL `json:"postgresql,omitempty"`
 
-	// Whether to migrate database on service startup?
+	// DB Migration approach for service?
+	// job: use job based approach
+	// service: service itself shall migrate the db
+	// none: no db migration needed
 	// +optional
-	// +default: false
-	MigrateDBOnStartUp bool `json:"migrateDBOnStartUp"`
+	// +kubebuilder:default:=service
+	DBMigrationApproach string `json:"dbMigrationApproach,omitempty"`
```

Comment:
```suggestion
	DBMigrationStrategy DBMigrationStrategyType  `json:"dbMigrationStrategy,omitempty"`
```
What's the default?

Please create the type and define the consts in this file:

```go
type DBMigrationStrategyType string

const (
    DBMigrationStrategyJob DBMigrationStrategyType = "job"
    DBMigrationStrategyService DBMigrationStrategyType = "service"
    DBMigrationStrategyNone DBMigrationStrategyType = "none"
)
```